### PR TITLE
Add GNU's include_next. Closes #104

### LINKED ISSUE #104: include_next
GNU extension:

https://gcc.gnu.org/onlinedocs/gcc-9.3.0/cpp/Wrapper-Headers.html#index-_0023include_005fnext

### PR DESCRIPTION
**Pull request by xquantxz**

### CODE REVIEW

**Comment by tyfkda** [comment]
@xquantxz Thank you to send this pull request!

Could you add unit test to `test_run()` function in cpptest.sh?
Why don't you add command line option `-I.` at line 40, or something.

**Comment by xquantxz** [comment]
> @xquantxz Thank you to send this pull request!
> 
> Could you add unit test to `test_run()` function in cpptest.sh?
> Why don't you add command line option `-I.` at line 40, or something.
> 

Is it okay if I test including stdlib.h?
else I would need to create a new directory to test on

**Comment by tyfkda** [comment]
> Is it okay if I test including stdlib.h?
> else I would need to create a new directory to test on

I prefer the latter.

File: src/cpp/preprocessor.c [hunk 1]
```diff
@@ -150,15 +172,18 @@ static void handle_include(const char *p, Stream *stream) {
   char *path = strndup(p, q - p);
   char *fn = NULL;
   FILE *fp = NULL;
+  char *dir = dirname(strdup(stream->filename));
```
Comment (tyfkda) [inline src/cpp/preprocessor.c:175]: I encountered test failed when I run the test on M1 Mac (which uses standard libc library on xcc).

It says `dirname()` function might return inner static buffer,
and it is overridden in second call in `fullpath()` called from `search_sysinc_next()`.

It looks wired, but please duplicate `direname`'s result:

```c
  char *dir = strdup(dirname(strdup(stream->filename)));
```

File: tests/cpptest.sh [hunk 2]
```diff
@@ -181,6 +181,11 @@ test_run() {
   try_run 'Comment after include' 73 "#include \"tmp.h\" /*block\n*/ // line\nint main(){return FOO;}"
   pp_error 'Token after include comment' "#include \"tmp.h\" /*block\n*/ illegal-token\nint main(){return FOO;}"
 
+  mkdir test_include
```
Comment (tyfkda) [inline tests/cpptest.sh:184]: Would you add `-p` option to suppress error on second time, please?

**Pull request merged**